### PR TITLE
LPD-28683 When clicking on order item without visibility the user is not redirected to the catalog page

### PR DESCRIPTION
**Pull request by monaco-fabio**
https://liferay.atlassian.net/browse/LPD-28683

### CODE REVIEW

**Comment by monaco-fabio** [comment]
ci:test:sf

**Comment by monaco-fabio** [comment]
ci:test:sf

File: modules/test/playwright/pages/commerce/commerce-order-content-web/pendingOrdersPage.ts [hunk 1]
```diff
@@ -36,6 +41,9 @@ export class PendingOrdersPage {
 			.getByTestId('specificationFacetPanel')
 			.getByRole('button');
 		this.viewButton = page.getByLabel('View');
+		this.showErrorMessage = page.getByText(
+			'One or more products are no longer available.'
+		);
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerce-order-content-web/pendingOrdersPage.ts:46]: No need to have the error message here.

File: modules/test/playwright/pages/commerce/commerce-order-content-web/pendingOrdersPage.ts [hunk 2]
```diff
@@ -16,8 +18,11 @@ export class PendingOrdersPage {
 	readonly pageTitle: Locator;
 	readonly panelList: Locator;
 	readonly viewButton: Locator;
+	readonly showErrorMessage: Locator;
 
 	constructor(page: Page) {
+		this.clickOnProductSku = (sku) => page.getByRole('link', {name: sku});
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerce-order-content-web/pendingOrdersPage.ts:24]: It is not an action, it should be `skuLink`

File: modules/test/playwright/pages/commerce/commerce-theme-minium/commerceThemeMiniumPage.ts [hunk 3]
```diff
@@ -8,12 +8,14 @@ import {Locator, Page} from '@playwright/test';
...
 	readonly page: Page;
 	readonly stickerUserNav: Locator;
+	readonly miniumLogo: Locator;
 	readonly myProfileItemMenu: Locator;
 
 	constructor(page: Page) {
 		this.page = page;
 
 		this.stickerUserNav = page.locator('.sticker').first();
+		this.miniumLogo = page.getByRole('link', {name: 'Go to Minium'});
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerce-theme-minium/commerceThemeMiniumPage.ts:18]: It is a link, so it should be called `goToMiniumLink`

File: modules/test/playwright/pages/commerce/commerceMiniCartPage.ts [hunk 4]
```diff
@@ -57,6 +59,9 @@ export class CommerceMiniCartPage {
 		this.unitOfMeasureTableLabel = page.getByText('Unit of Measure Table', {
 			exact: true,
 		});
+		this.viewOrderDetails = page.getByRole('button', {
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerceMiniCartPage.ts:62]: It is a button, so it should be `viewDetailsButton`

File: modules/test/playwright/pages/commerce/commerceMiniCartPage.ts [hunk 5]
```diff
@@ -66,4 +71,12 @@ export class CommerceMiniCartPage {
 		await this.quickAddToCartButton.click();
 		await this.miniCartButtonClose.click();
 	}
+
+	async goToPeningOrder(sku: string) {
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerceMiniCartPage.ts:75]: Fix typo `goToPeningOrder` -> `goToPendingOrder`

File: modules/test/playwright/pages/commerce/commerceMiniCartPage.ts [hunk 6]
```diff
@@ -66,4 +71,12 @@ export class CommerceMiniCartPage {
...
+		await this.miniCartButton.click();
+		await this.searchProductsInput.fill(sku);
+		await this.searchProductsInput.waitFor();
+		await this.quickAddToCartSku(sku).click();
+		await this.quickAddToCartButton.click();
+	}
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerceMiniCartPage.ts:81]: It seems that this method does another thing

File: modules/test/playwright/pages/users-admin-web/EditUserPage.ts [hunk 7]
```diff
@@ -19,16 +20,28 @@ export class EditUserPage {
...
 	readonly selectOrganizationsTableRow: (
 		colPosition: number,
 		value: string,
 		strictEqual?: boolean
 	) => Promise<{column: Locator; row: Locator}>;
+	readonly selectMembershipSite: Locator;
 	readonly webDAVPasswordLabel: Locator;
 
 	constructor(page: Page) {
+		this.frameSelectSite = async (siteName: string) => {
```
Comment (alessiorendina) [inline modules/test/playwright/pages/users-admin-web/EditUserPage.ts:35]: It is a link, so it should be `selectSiteLink`

File: modules/test/playwright/pages/users-admin-web/EditUserPage.ts [hunk 8]
```diff
@@ -19,16 +20,28 @@ export class EditUserPage {
...
+			return page
+				.frameLocator('iframe[title="Select Site"]')
+				.getByRole('link', {exact: true, name: siteName});
+		};
+
+		this.saveMemberSiteButton = page.getByRole('button', {name: 'Save'});
```
Comment (alessiorendina) [inline modules/test/playwright/pages/users-admin-web/EditUserPage.ts:41]: This is not specific for the member site, so it should be `saveButton`

File: modules/test/playwright/pages/users-admin-web/EditUserPage.ts [hunk 9]
```diff
@@ -19,16 +20,28 @@ export class EditUserPage {
...
+
+		this.selectMembershipSite = page.getByLabel('Select Sites');
```
Comment (alessiorendina) [inline modules/test/playwright/pages/users-admin-web/EditUserPage.ts:43]: It is too generic, can we locate it better?

File: modules/test/playwright/pages/users-admin-web/EditUserPage.ts [hunk 10]
```diff
@@ -86,4 +99,12 @@ export class EditUserPage {
 			);
 		};
 	}
+
+	async assignUserToSite(site: string) {
```
Comment (alessiorendina) [inline modules/test/playwright/pages/users-admin-web/EditUserPage.ts:103]: `selectUserMembershipSite`

**Review by alessiorendina — CHANGES_REQUESTED** [review]
@monaco-fabio please check my comments, thank you!

**Comment by monaco-fabio** [comment]
ci:test:sf

File: modules/test/playwright/pages/commerce/commerce-order-content-web/pendingOrdersPage.ts [hunk 11]
```diff
@@ -8,6 +8,8 @@ import {Locator, Page} from '@playwright/test';
 import {CommerceLayoutsPage} from '../commerceLayoutsPage';
 
 export class PendingOrdersPage {
+	readonly clickOnProductSkuLink: (sku: string) => Locator;
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerce-order-content-web/pendingOrdersPage.ts:11]: skuLink

File: modules/test/playwright/pages/commerce/commerce-order-content-web/pendingOrdersPage.ts [hunk 12]
```diff
@@ -16,8 +18,12 @@ export class PendingOrdersPage {
 	readonly pageTitle: Locator;
 	readonly panelList: Locator;
 	readonly viewButton: Locator;
+	readonly showErrorMessage: Locator;
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerce-order-content-web/pendingOrdersPage.ts:21]: This needs to be removed

File: modules/test/playwright/pages/commerce/commerce-order-content-web/pendingOrdersPage.ts [hunk 13]
```diff
@@ -8,6 +8,8 @@ import {Locator, Page} from '@playwright/test';
 import {CommerceLayoutsPage} from '../commerceLayoutsPage';
 
 export class PendingOrdersPage {
+	readonly clickOnProductSkuLink: (sku: string) => Locator;
+	readonly closeErrorMessage: Locator;
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerce-order-content-web/pendingOrdersPage.ts:12]: errorMessageCloseButton

File: modules/test/playwright/pages/commerce/commerceMiniCartPage.ts [hunk 14]
```diff
@@ -28,6 +28,7 @@ export class CommerceMiniCartPage {
 	readonly showOptionsButton: Locator;
 	readonly submitButton: Locator;
 	readonly unitOfMeasureTableLabel: Locator;
+	readonly viewOrderDetailsButton: Locator;
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerceMiniCartPage.ts:31]: viewDetailsButton

File: modules/test/playwright/pages/commerce/commerceMiniCartPage.ts [hunk 15]
```diff
@@ -66,13 +68,17 @@ export class CommerceMiniCartPage {
 		this.unitOfMeasureTableLabel = page.getByText('Unit of Measure Table', {
 			exact: true,
 		});
+		this.viewOrderDetailsButton = page.getByRole('button', {
+			name: 'View Details',
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerceMiniCartPage.ts:72]: Add `exact: true`

File: modules/test/playwright/pages/users-admin-web/EditUserPage.ts [hunk 16]
```diff
@@ -31,10 +31,13 @@ export class EditUserPage {
 		value: string,
 		strictEqual?: boolean
 	) => Promise<{column: Locator; row: Locator}>;
+	readonly membershipSiteSelectButton: Locator;
```
Comment (alessiorendina) [inline modules/test/playwright/pages/users-admin-web/EditUserPage.ts:34]: Check the other PR

File: modules/test/playwright/tests/commerce/commerce-order-content-web/pendingOrder.spec.ts [hunk 17]
```diff
@@ -391,3 +394,131 @@ test('LPD-4174 Sales agent can receive email notifications for new orders placed
...
+	apiHelpers.data.push({id: site.id, type: 'site'});
+
+	const account = await apiHelpers.headlessAdminUser.postAccount({
+		name: 'admin',
+		type: 'business',
+	});
+
+	apiHelpers.data.push({id: account.id, type: 'account'});
+
+	await commerceLayoutsPage.cleanupSiteInitializerData(apiHelpers, site.name);
```
Comment (alessiorendina) [inline modules/test/playwright/tests/commerce/commerce-order-content-web/pendingOrder.spec.ts:423]: Let's move this after site creation or in a finally block

File: modules/test/playwright/tests/commerce/commerce-order-content-web/pendingOrder.spec.ts [hunk 18]
```diff
@@ -391,3 +394,131 @@ test('LPD-4174 Sales agent can receive email notifications for new orders placed
...
+	);
+
+	await commerceMiniCartPage.viewOrderDetailsButton.click();
+
+	await expect(
+		page.getByText('One or more products are no longer available.')
+	).toBeVisible();
+
+	await pendingOrdersPage.closeErrorMessage.click();
+
```
Comment (alessiorendina) [inline modules/test/playwright/tests/commerce/commerce-order-content-web/pendingOrder.spec.ts:518]: Remove empty line

**Review by alessiorendina — CHANGES_REQUESTED** [review]
@monaco-fabio please check my comments, thanks!

**Comment by monaco-fabio** [comment]
ci:test:sf

File: modules/test/playwright/pages/commerce/commerce-order-content-web/pendingOrdersPage.ts [hunk 19]
```diff
@@ -8,6 +8,8 @@ import {Locator, Page} from '@playwright/test';
 import {CommerceLayoutsPage} from '../commerceLayoutsPage';
 
 export class PendingOrdersPage {
+	readonly skuLink: (sku: string) => Locator;
+	readonly errorMessageCloseButton: Locator;
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerce-order-content-web/pendingOrdersPage.ts:12]: Sort alphabetically

File: modules/test/playwright/pages/commerce/commerce-theme-minium/commerceThemeMiniumPage.ts [hunk 20]
```diff
@@ -8,12 +8,14 @@ import {Locator, Page} from '@playwright/test';
...
 	readonly stickerUserNav: Locator;
+	readonly goToMiniumLink: Locator;
 	readonly myProfileItemMenu: Locator;
 
 	constructor(page: Page) {
 		this.page = page;
 
 		this.stickerUserNav = page.locator('.sticker').first();
+		this.goToMiniumLink = page.getByRole('link', {name: 'Go to Minium'});
 		this.myProfileItemMenu = page.getByRole('link', {name: 'My Profile'});
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerce-theme-minium/commerceThemeMiniumPage.ts:19]: Sort alphabetically, remove empty line

File: modules/test/playwright/pages/commerce/commerceMiniCartPage.ts [hunk 21]
```diff
@@ -27,6 +27,7 @@ export class CommerceMiniCartPage {
 	readonly searchProductsInput: Locator;
 	readonly showOptionsButton: Locator;
 	readonly submitButton: Locator;
+	readonly viewDetailsButton: Locator;
 	readonly unitOfMeasureTableLabel: Locator;
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerceMiniCartPage.ts:32]: Sort alphabetically

**Review by alessiorendina — CHANGES_REQUESTED** [review]
@monaco-fabio please check my comments, thank you!

**Comment by monaco-fabio** [comment]
ci:test:sf

**Comment by monaco-fabio** [comment]
ci:test:playwright:nocompile

File: modules/test/playwright/pages/commerce/commerce-theme-minium/commerceThemeMiniumPage.ts [hunk 22]
```diff
@@ -8,12 +8,13 @@ import {Locator, Page} from '@playwright/test';
 export class CommerceThemeMiniumPage {
 	readonly page: Page;
 	readonly stickerUserNav: Locator;
+	readonly goToMiniumLink: Locator;
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerce-theme-minium/commerceThemeMiniumPage.ts:11]: Sort alphabetically

**Review by alessiorendina — CHANGES_REQUESTED** [review]
@monaco-fabio please check my comment and make sure the order is deleted as well. Thank you!

**Comment by monaco-fabio** [comment]
ci:test:sf

**Pull request closed without merging**